app/services/cascade_assets_client.rb

```diff
@@ -0,0 +1,153 @@
+    # Savon ignores this ordering when it returns the response, so for now we need to rebuild the request hash from the response putting values
+    # in the order defined in the WSDL.
+    # There is an open issue for this: https://github.com/savonrb/savon/issues/592
+    asset = {
+      xmlBlock: {
+        id: xml_block[:id],
+        name: xml_block[:name],
+        parentFolderId: xml_block[:parent_folder_id],
+        siteName: xml_block[:site_name],
+        xml: xml_block[:xml]
+      }
+    }
+
+    message = {authentication: deploy_credentials, asset: asset}
+    request(:edit, message)
+  end
+
+  # Currently there is no way to upload asset files without duplicating the ones
+  # that already exist (see https://help.hannonhill.com/hc/en-us/requests/4460)
+  # So instead, delete all assets files before recreating them in the upload method
+  def delete_assets
+    batch_request = {
+      authentication: deploy_credentials,
+      operation: [
+        {
+          delete: {
+            identifier: {
+              path: {
+                path: CASCADE_CONFIG['asset_path'],
+                siteName: CASCADE_CONFIG['site_name']
+              },
+              type: "folder"
+            }
+          },
+        },
+        create: folder_asset(CASCADE_CONFIG['asset_path'])
+      ]
+    }
+
+    request(:batch, batch_request)
+  end
+
+  def upload_assets
+    local_assets, operations = [], []
+
+    Dir.chdir(local_asset_path) do
+      local_assets = Dir.glob("**/*", File::FNM_DOTMATCH).reject{|asset| File.basename(asset) == "." || File.basename(asset) == ".."}
+    end
+
+    local_assets.each do |path|
+      remote_path, local_path = remote_asset_path(path), local_asset_path(path)
+      asset = File.directory?(local_path) ? folder_asset(remote_path) : file_asset(remote_path, IO.read(local_path))
+      operations.push({create: asset})
+    end
+
+    batch_request = {
+      authentication: deploy_credentials,
+      operation: operations
+    }
+
+    request(:batch, batch_request)
+  end
+
+  def file_asset(remote_path, data)
+    encoded_data = Base64.encode64(data)
+
+    {
+      asset: {
+        file: {
+          name: File.basename(remote_path),
+          parentFolderPath: parent_dir_path(remote_path),
+          siteName: CASCADE_CONFIG['site_name'],
+          metadataSetPath: '/Default',
+          shouldBePublished: 'false',
+          shouldBeIndexed: 'false',
+          data: encoded_data,
+          rewriteLinks: 'false',
+          maintainAbsoluteLinks: 'false'
+        }
+      }
+    }
+  end
+
+  def folder_asset(remote_path)
+    {
+      asset: {
+        folder: {
+          name: File.basename(remote_path),
+          parentFolderPath: parent_dir_path(remote_path),
+          siteName: CASCADE_CONFIG['site_name'],
+          metadataSetPath: '/Website Folder'
+        }
+      }
+    }
+  end
```

Comment:
Because this was just a spike, I did not do this in the prettiest way. Eventually it would probably make sense to make an `Asset` class rather than having `file_asset`, `folder_asset`, etc..